api_server/src/request/mod.rs

```diff
@@ -98,11 +98,11 @@ impl PartialEq for ParsedRequest {
 
 #[cfg(test)]
 mod tests {
-    extern crate arch;
-    extern crate devices;
-    extern crate kernel;
-    extern crate memory_model;
-    extern crate net_util;
+    use arch;
```

Comment:
These uses are not needed, please delete them.

The code can be simplified by importing what is actually needed from those crates by using the crate name directly instead of `self`.
```
    use devices::virtio::net::Error as VirtioNetError;
    use memory_model::GuestMemoryError;
    use net_util::TapError;
```